auth/templates/actions/auth.go.tmpl

```diff
@@ -55,7 +55,12 @@ func AuthCreate(c buffalo.Context) error {
 	c.Session().Set("current_user_id", u.ID)
 	c.Flash().Add("success", "Welcome Back to Buffalo!")
 
-	return c.Redirect(302, "/")
+	redirectURL := "/"
+	if c.Session().Get("redirectURL") != nil {
```

Comment:
Instead of checking nil do a type assert instead. You can make this cleaner and safer. Currently the next line will panic if the value is anything other than a string.